iceoryx_hoofs/include/iceoryx_hoofs/design_pattern/builder_pattern.hpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_DESIGN_PATTERN_BUILDER_PATTERN_HPP
```

Comment:
This is quite inconsistent. For example the `NewType` is in `cxx`. Could this maybe be combined with another module?

Reply:
@elBoberido I could but since the design pattern folder already existed I thought it was the perfect place. What do you think of moving `NewType` into this folder as well?

In the end they both do not belong into `cxx`.